test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -727,6 +772,91 @@ public void CreateDisplayMetadata_IsFlagsEnum_ReflectsModelType(Type type, bool
                 .ThenBy(item => item.Key.Name, StringComparer.Ordinal));
         }
 
+        private DataAnnotationsMetadataProvider CreateLocalizingProvider()
+        {
+            var stringLocalizer = new Mock<IStringLocalizer>(MockBehavior.Strict);
+            stringLocalizer
+                .Setup(loc => loc["LOC_Two"])
+                .Returns<string>((k => new LocalizedString(k, $"{k} {CultureInfo.CurrentCulture}")));
+
+            var stringLocalizerFactory = new Mock<IStringLocalizerFactory>(MockBehavior.Strict);
+            stringLocalizerFactory
+                .Setup(factory => factory.Create(typeof(EnumWithLocalizedDisplayNames)))
+                .Returns(stringLocalizer.Object);
+
+            return new DataAnnotationsMetadataProvider(stringLocalizerFactory: stringLocalizerFactory.Object);
+        }
+
+        [Fact]
+        public void CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_IStringLocalizer()
+        {
+            // Arrange
+            var provider = CreateLocalizingProvider();
+
+            var key = ModelMetadataIdentity.ForType(typeof(EnumWithLocalizedDisplayNames));
+            var attributes = new object[0];
+
+            // Act
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+            provider.CreateDisplayMetadata(context);
+
+            string frenchEnumDisplay;
+            using (new CultureReplacer("fr-FR", "fr-FR"))
+            {
+                frenchEnumDisplay = context.DisplayMetadata.EnumGroupedDisplayNamesAndValues
+                    .Where(kvp => kvp.Value == "2")
+                    .First().Key.Name;
+            }
+
+            string englishEnumDisplay;
+            using (new CultureReplacer("en-US", "en-US"))
+            {
+                englishEnumDisplay = context.DisplayMetadata.EnumGroupedDisplayNamesAndValues
+                    .Where(kvp => kvp.Value == "2")
+                    .First().Key.Name;
+            }
+
+            // Assert
+            Assert.NotEqual(frenchEnumDisplay, englishEnumDisplay);
+            Assert.Equal("LOC_Two fr-FR", frenchEnumDisplay);
+            Assert.Equal("LOC_Two en-US", englishEnumDisplay);
+        }
+
+        [Fact]
+        public void CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_ResourceTypeLocalizes()
```

Comment:
Add similar dual-locale tests for `Description`, `DisplayName`, and `Placeholder`. Now that you've fixed `TestResources`, should work w/ either an additional `.resx` for another locale or more properties like `Type_Three`. Up to you.

I'll admit these additional tests aren't about a specific bug. But the test gap you identified exists there as well.